src/api/v1beta1/properties.go

```diff
@@ -387,3 +406,30 @@ func splitArg(arg string) (key, value string) {
 	value = split[1]
 	return
 }
+
+func getRawImageTag(imageURI string) string {
+	if !strings.Contains(imageURI, ":") {
+		return "latest"
+	}
+	splitURI := strings.Split(imageURI, ":")
+	return splitURI[len(splitURI)-1]
+}
+
+// getSafeImageTag extracts and normalizes the provided image's tag
+func getSafeImageTag(imageURI string) string {
+	rawTag := getRawImageTag(imageURI)
+	return normalizeImageTag(rawTag)
+}
+
+// normalizeImageTag normalizes the tag
+// so it matches the requirements described here: https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/#syntax-and-character-set
+func normalizeImageTag(tag string) string {
+	notAllowedCharsRegex := regexp.MustCompile(`[^\d\w-_.]`)
+	safeTag := notAllowedCharsRegex.ReplaceAllString(tag, "-")
+	if len(safeTag) > maxLabelValueLength {
+		safeTag = safeTag[:maxLabelValueLength]
+	}
+	notAllowedEndingRegex := regexp.MustCompile(`[^\w\d]+$`)
+	safeTag = notAllowedEndingRegex.ReplaceAllString(safeTag, "")
+	return safeTag
+}
```

Comment:
I think we can find a library to do it, e.g.: https://pkg.go.dev/github.com/novln/docker-parser#section-readme
Less code to maintain (prod and tests).

Reply:
Or actually this is what you need: https://github.com/moby/moby/blob/49bf474f9ed7ce7143a59d1964ff7b7fd9b52178/api/types/reference/image_reference_test.go
Moby is Docker's project :+1:.

Reply:
First one is made by a dude with the only release being two years old, the second one would only retrieve the tag but not sanitize it. But since I just noticed it, those functions would be better kept in a separate package instead.

Reply:
separate external library 👎
separate package  👍

Reply:
Us maintaining regexes :-1: 
Names collision :-1: :smile: 
https://github.com/Dynatrace/dynatrace-operator/pull/843#discussion_r908179852